apps/src/templates/teacherDashboard/SectionsPage.jsx

```diff
@@ -106,7 +110,22 @@ class SectionsPage extends Component {
     });
   };
 
-  addSection = () => this.props.newSection();
+  handleCloseAddSectionDialogs = () => {
+    this.setState({addSectionDialogOpen: false});
+  };
+
+  addSection = () => {
+    if (experiments.isEnabled('section-flow-2017')) {
+      this.setState({addSectionDialogOpen: true});
+    } else {
+      return this.props.newSection();
+    }
+  };
+
+  handleEditRequest = (section) => {
+    this.setState({editSectionDialogOpen : true});
+    this.editer.getWrappedInstance().updateStates(section);
```

Comment:
I wonder a little bit if this is another instance of where we'd be better off moving state to redux and thus not need to intertwine the refs